library/src/main/java/com/getstream/sdk/chat/navigation/destinations/ChatDestination.kt

```diff
@@ -0,0 +1,17 @@
+package com.getstream.sdk.chat.navigation.destinations
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+
+abstract class ChatDestination(protected var context: Context) {
+    abstract fun navigate()
+
+    protected fun start(intent: Intent) {
+        context.startActivity(intent)
+    }
+
+    protected fun startForResult(intent: Intent, requestCode: Int) {
+        (context as Activity).startActivityForResult(intent, requestCode)
```

Comment:
This casting could fails, not all `Context` are `Activity`, we should handle it

Reply:
True, added a more explicit check and a nicely worded exception message for this